### ISSUE TITLE
[FR] Native method to tell if a command error has been handled in a command-specific error handler from the global 

### ISSUE THREAD

**Issue by Vexs** [Rapptz/discord.py#1985]
### The Problem

It's not uncommon to have error strings that only apply to a specific command- ie, in a tag system, you'd respond "tag name not specified" if no arguments are provided. This string obviously wouldn't make sense to have for other commands, so it makes sense to handle in a command-local error handler. However, if you have a generic missing argument handler in your global, it will also respond.

### The Ideal Solution

The best solution I can think of is to have a flag you can set on the ctx object that prevents other error handlers from being dispatched. Frankly, I don't really like this.

### The Current Solution

General solution is to use a custom context object with the flag, then check if the attribute is set in on_command_error- if it is, return. Another possibility is to check what command is originating the specific error, and check against some blacklist in the global on_command_error. However, this feels dirty and nonmodular.

### Summary

In short, some method to tell if a command error has been handled in a command-local error handler before raising to the global on_command_error- the above methods are effective, but they feel dirty to me. I have no idea how to implement this cleanly, which is why I raise an issue instead.

**Comment by Rapptz** [comment]
I will just repeat what I said in the official server and place it here for future purposes.

There are two main issues with this request long term.

1) There is no `__slots__` set on `Context`, so the current and ideal solution you propose are both possible to do today with no extra work on my part. 
2) There is no way for me to (currently) tell if something is actually "handled". Let me go over the different definitions of what "being handled" currently mean and their problems.

1. The first and easiest way to check if something is handled is through seeing if a local error handler exists. This is what the default `on_command_error` does.  However in reality, it probably isn't handled. Just because we have a local error handler that handles say `BadArgument` doesn't mean we want it to not propagate and let some other cog local or global event handler have its turn. So clearly this one's out.
2. Returning a truth-y value from the function will prevent propagation. 
3. Add a `stop_propagation` method to `Context` (a la JavaScript). I'm going to tackle both 2 and 3 here because both of these are essentially the same, just different in implementation. The main problem with this solution as far as I can tell is that certain logging cogs become useless. If you have a cog that logs errors for whatever reason and another error handler has stopped the propagation then suddenly you're missing some important logs. There is no way to make everyone happy in these scenarios.

Quite frankly, the design of the exception hierarchy should ideally make it so certain ones are caught in a local way (e.g. `BadArgument`, some custom `CommandError`, etc) while others should be handled at the global scale (e.g. `ArgumentParsingError`, `CommandNotFound`, `DisabledCommand`, etc).
